### PR TITLE
Add better dependencies for generated files

### PR DESCRIPTION
**Pull request by levitte**
Most of all, `Makefile` is added as a dependency, because we sometimes
make changes there that should affect re-generation.  The recent change
of how `community/otc.inc` is generated is the perfect example.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -282,24 +282,24 @@ policies/glossary.html: $(GLOSSARY) bin/md-to-html5 Makefile
...
 	./bin/mk-omc -n -p -t 'OTC Members' otc otc-inactive > $@
-community/omc.inc: $(PERSONDB)
+community/omc.inc: $(PERSONDB) bin/mk-omc Makefile
 	./bin/mk-omc -n -e -l -p -t 'OMC Members' omc omc-inactive > $@
-community/omc-alumni.inc: $(PERSONDB)
+community/omc-alumni.inc: $(PERSONDB) bin/mk-omc Makefile
 	./bin/mk-omc -n -l -t 'OMC Alumni' omc-alumni omc-emeritus > $@
 
-news/changelog.inc: news/changelog.txt bin/mk-changelog
+news/changelog.inc: news/changelog.txt bin/post-process-html5 Makefile
```
Comment (paulidale) [inline Makefile:302]: Is losing bin/mk-changelog a problem here?

Comment (levitte) [reply Makefile:302]: If you look at the commands, you will see that it's not used anywhere any more.  We could actually drop `bin/mk-changelog` altogether, 'cause we don't use it any more...

Comment (levitte) [reply Makefile:302]: That happened in commit 4b0220368e888aab29972537aff8602a45b724e9

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged

f16ed1e69f3808c79b223a4117bb340619559a7b Add better dependencies for generated files

**Comment by levitte** [comment]
Do note that this didn't contain the dependency on `dirdata.yaml` for
.html files generated from .d files.  That's done by #326.

File: Makefile [hunk 2]
```diff
@@ -381,10 +381,10 @@ news/newsflash.inc: news/newsflash.txt
 #
 # $(1) = output file mod, $(2) = release version switch, $(3) = release version
 define mknews_vulnerability
-news/vulnerabilities$(1).inc: bin/mk-cvepage news/vulnerabilities.xml
+news/vulnerabilities$(1).inc: bin/mk-cvepage news/vulnerabilities.xml Makefile
 	@rm -f $$@
 	./bin/mk-cvepage -i news/vulnerabilities.xml $(2) > $$@
-news/vulnerabilities$(1).html: news/vulnerabilities.html.tt bin/from-tt
+news/vulnerabilities$(1).html: news/vulnerabilities.html.tt bin/from-tt Makeflie
```
Comment (levitte) [inline Makefile:387]: Oops, spelling!  Fixed in #331

**Pull request closed without merging**